modules/server/src/test/scala/scaladex/server/route/ControllerBaseSuite.scala

```diff
@@ -22,7 +29,26 @@ trait ControllerBaseSuite extends AsyncFunSpec with Matchers with ScalatestRoute
     realConfig.copy(filesystem = realConfig.filesystem.copy(index = index))
   }
 
-  val githubUserSession = new GithubUserSession(config.session)
+  private implicit val cs: ContextShift[IO] =
+    IO.contextShift(ExecutionContext.global)
+
+  private val databaseConfig: PostgreSQLConfig = PostgreSQLConfig
+    .load()
+    .get
+    .asInstanceOf[PostgreSQLConfig]
+
+  val transactor: Transactor.Aux[IO, Unit] =
+    Transactor
+      .fromDriverManager[IO](
+        databaseConfig.driver,
+        databaseConfig.url,
+        databaseConfig.user,
+        databaseConfig.pass.decode
+      )
+
+  val sqlDatabase = new SqlDatabase(databaseConfig, transactor)
```

Comment:
@adpi2 

Looking into the [failing unit tests](https://github.com/scalacenter/scaladex/runs/5727364612?check_suite_focus=true), I think it might be because of these lines.

I copied over this logic from here

https://github.com/scalacenter/scaladex/blob/86357e5846f7e79448d80d50a549e3a250e3db46/modules/infra/src/test/scala/scaladex/infra/BaseDatabaseSuite.scala#L18-L19

Because I didn't think extending this trait in a test suite that's not directly testing the database was a wise choice.

Reply:
Ok I was surprised by the failing tests but now I understand better. The problem is that we don't start the PostgreSQL container when running `server / test`.

The current design for the test is:
- the database and other third-parties (`GithubClient`, `ElasticSearchEngine`...) are unit-tested in `infra / test` and `infra / IntegrationTest / test`
- we use some mock implementations in `server / test`

The mock for the database is `InMemoryDatabase`. The mock for the `GithubAuth` is `MockGithubAuth`. I think you should be able to run all the tests in `server / test` by adapting those mocks instead of plugging in a real database and calling the real Github API.

Reply:
You should see my suggestion below